tkey/src/main/java/com/web3auth/tkey/ThresholdKey/Modules/TSSModule.java

```
@@ -0,0 +1,510 @@
+        pubObject.put("y", publicAddress.publicKey.y);
+        JSONArray nodeIndexArray = new JSONArray();
+        for (BigInteger nodeIndex: publicAddress.nodeIndexes) {
+            nodeIndexArray.put(nodeIndex);
+        }
+        JSONObject jsonPubKey = new JSONObject();
+
+        jsonPubKey.put("nodeIndexes", nodeIndexArray);
+        jsonPubKey.put("publicKey", pubObject);
+
+        setTSSTag(thresholdKey, tssTag, result -> {
+            if (result instanceof Result.Error) {
+                throw new RuntimeException("failed to set TSS Tag");
+            }
+            thresholdKey.executor.execute(() -> {
+                try {
+                    Result<Boolean> updateResult = updateTssPubKey(thresholdKey, tssTag, String.valueOf(nonce), jsonPubKey.toString());
+                    callback.onComplete(updateResult);
+                } catch (Exception e) {
+                    Result<Boolean> error2 = new Result.Error<>((Exception) e);
+                    callback.onComplete(error2);
+                }
+            });
+        });
+    }
+
+    private static Result<Boolean> updateTssPubKey(ThresholdKey thresholdKey, String tssTag, String nonce, String pubKey) {
+        try {
+            RuntimeError error = new RuntimeError();
+            jniThresholdKeyServiceProviderAssignPublicKey(thresholdKey, tssTag, nonce, pubKey, error);
+            if (error.code != 0) {
+                throw error;
+            }
+            return new Result.Success<>();
+        } catch (Exception | RuntimeError e) {
+            return new Result.Error<>((Exception) e);
+        }
+    }
+
+    public static void AddFactorPub(ThresholdKey thresholdKey, String TSSTag, String factorKey, ArrayList<String> authSignatures, String newFactorPub, int newTssIndex, @Nullable int[] selectedServers, NodeDetails nodeDetails, TorusUtils torusUtils, ThresholdKeyCallback<Boolean> callback) throws RuntimeError, Exception {
+        if (factorKey.length() > 66) {
```

check if this is wrappable to an executor, else check errors are not being swallowed and execution occurs serially, throwing on error at each step